packages/devtools_app/lib/src/inspector/flutter/inspector_screen_details_tab.dart

```diff
@@ -101,67 +101,28 @@ class _LayoutDetailsTabState extends State<LayoutDetailsTab>
 
   RemoteDiagnosticsNode get selected => controller?.selectedNode?.diagnostic;
 
-  InspectorObjectGroupManager objectGroupManager;
+  RemoteDiagnosticsNode previousSelection;
 
-  RemoteDiagnosticsNode root;
-
-  RemoteDiagnosticsNode getRoot(RemoteDiagnosticsNode node) {
-    if (!StoryOfYourFlexWidget.shouldDisplay(node)) return null;
-    if (node.isFlex) return node;
-    return node.parent;
+  Widget responsibleWidget(RemoteDiagnosticsNode node) {
+    if (StoryOfYourFlexWidget.shouldDisplay(node))
+      return StoryOfYourFlexWidget(controller);
+    return const SizedBox();
   }
 
-  void onSelectionChanged() async {
-    if (!StoryOfYourFlexWidget.shouldDisplay(selected)) {
-      setState(() => root = null);
-      return;
-    }
-    final shouldFetch =
-        root?.dartDiagnosticRef?.id != getRoot(selected)?.dartDiagnosticRef?.id;
-    if (shouldFetch) {
-      objectGroupManager.cancelNext();
-      // TODO(albertusangga) show loading animation when root is null?
-      setState(() {
-        root = null;
-      });
-
-      final nextObjectGroup = objectGroupManager.next;
-      root = await nextObjectGroup.getDetailsSubtreeWithRenderObject(
-        selected.isFlex ? selected : selected.parent,
-        subtreeDepth: 1,
-      );
-      if (!nextObjectGroup.disposed) {
-        assert(objectGroupManager.next == nextObjectGroup);
-        objectGroupManager.promoteNext();
-      }
+  void onSelectionChanged() {
+    if (responsibleWidget(previousSelection).runtimeType !=
+        responsibleWidget(selected).runtimeType) {
+      setState(() {});
     }
-    setState(() {});
+    previousSelection = selected;
```

Comment:
This should go in a setState call.

Reply:
In the case where the responsible widget is not changed I don't want to call the setState though. I will wrap this inside `else` block then.